tests/e2e/specs/settings/applications-passwords.test.js

```diff
@@ -0,0 +1,103 @@
+import {
+    visitAdminPage,
+    __experimentalRest as rest,
+} from '@wordpress/e2e-test-utils';
+
+async function getResponseForApplicationPassword() {
+    return await rest({
+        method: 'GET',
+        path: '/wp/v2/users/me/application-passwords'
+    });
+}
+
+async function createApplicationPassword(applicationName) {
+    await visitAdminPage('profile.php');
+    await page.waitForSelector('#new_application_password_name');
+    await page.type('#new_application_password_name', `${applicationName}`);
+    await page.click('#do_new_application_password');
+
+    await page.waitForSelector('.notice');
+}
+
+async function revokeAllApplicationPasswords() {
+    await visitAdminPage('profile.php');
+
+    const revokeAllButtonVisibility = await page.evaluate(() => {
+        const e = document.querySelector('.application-passwords-list-table-wrapper');
+        if (!e) {
+            return false;
+        }
+        const visibility = window.getComputedStyle(e);
+        return visibility && visibility.display !== 'none' && visibility.visibility !== 'hidden' && visibility.opacity !== '0';
+    });
+
+    if (revokeAllButtonVisibility) {
+        await page.click('#revoke-all-application-passwords');
+        await page.keyboard.press('Enter');
+        await page.waitForSelector('.notice-success');
+    }
+}
+
+describe('Manage applications passwords', () => {
+    const testApplicationName = 'Test Application';
+
+    beforeEach(async() => {
+        await revokeAllApplicationPasswords();
```

Comment:
While we do need a test to _manually_ revoke all passwords, I don't think we have to do it for every test just to clear the states. Is it possible to call APIs to clear the states here instead of repeating this for every test? It could potentially speed up our tests too.

Reply:
Yeah, interesting. It saves 1 to 1.5 seconds for the whole test.

How far we want to go with optimizations? I guess we could try the same thing with creating a password? As the creation feature is already tested.